pkg/controller/controller_crd_event_handler.go

```diff
@@ -25,29 +25,21 @@ type watchState struct {
 type crdEventHandler struct {
 	ctx context.Context
 	*BundleController
-	mx       sync.Mutex            // protects the map
 	watchers map[string]watchState // CRD name -> state
 }
 
 func (h *crdEventHandler) OnAdd(obj interface{}) {
 	crd := obj.(*apiext_v1b1.CustomResourceDefinition)
-	if crd.Name == smith.BundleResourceName {
-		return
+	if h.ensureWatch(crd) {
```

Comment:
Is this meant to address the 'reacting before API has it available' issue? If so, I don't understand - it seems like we're just erroring out more cleanly, but we'll still miss establishing the watch later on?